### ISSUE TITLE
Can getters and setters be added to a class?

### ISSUE THREAD

**Issue by danimbrogno** [dsherret/ts-morph#123]
Can't figure out how to add a getter or setter method to a class, the following throws a not implemented error.

```
return file.addClass(
      {
        name: this.namer.segmentHeadRowToClassName(segment),
        isExported: true,
        methods: [
          {
            name: 'get rawSegment',
            bodyText: `return [this._segmentId, ...this._elements];`
          }
        ],

      }
    );
```

returns: 

`Unexpected! Inserting syntax kind of MethodDeclaration, but GetAccessor was inserted.`

 Is this not possible at this time? Is it likely to be added in the future? Any work arounds?

**Comment by dsherret** [comment]
Yeah, not possible at this time. I'm working on it in #27. I'll try to speed up the work on that.

**Good Workaround**

I just realized the library doesn't support this, but I'll do a commit and release tonight so it does. Basically you will be able to get around it by adding the class, then inserting a statement for the get accessor:

```typescript
const classDeclaration = file.addClass({ ...etc... });
classDeclaration.insertStatements(index, writer => {
    writer.writeLine("get rawSegment()").block(() => {
        writer.writeLine("`return [this._segmentId, ...this._elements];`");
    });
    writer.writeLine(); // maybe add a newline here? or at the beginning... this is untested
});
```

Alternatively instead of using a writer, you could just provide a string.

```typescript
classDeclaration.insertStatements(0, "get rawSegment() {\n    return [this._segmentId, ...this._elements];\n}\n");
```

**Bad Workaround**

This currently works. Insert it using `sourceFile.insertText(pos, text)`... read about why using this isn't so great here: https://dsherret.github.io/ts-simple-ast/manipulation/

**Comment by dsherret** [comment]
@danimbrogno the "good workaround" I describe above is implemented in 0.89.0. It's somewhat hidden and undocumented because I only want it to be used for workarounds at the current moment.

**Comment by danimbrogno** [comment]
The workaround works great, thanks for the quick turnaround! I'll keep an eye out for the official implementation.

**Comment by dsherret** [comment]
This is implemented in 0.90.0. Use `classDeclaration.addGetAccessor(...);`, `classDeclaration.addSetAccessor(...)` and similar methods.
